html/semantics/document-metadata/interactions-of-styling-and-scripting/conditionally-block-rendering-on-link-media-attr.html

```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+
+<link rel=stylesheet href=stylesheet.py>
+<link rel=stylesheet media="screen and (max-width:10px)" href=stylesheet.py?stylesNotMatchingEnvironment&delay=2>
+<h1>Dominic Farolino</h1>
+<script>
+  function styleExists(styleText) {
+    for (let key in document.styleSheets) {
+      if (!document.styleSheets.hasOwnProperty(key) || key == "length") continue;
+
+      let currentStyleText = document.styleSheets[key].cssRules["0"].cssText;
+      if (currentStyleText == styleText) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  test(() => {
+    const h1 = document.querySelector('h1');
+    const computedColor = getComputedStyle(h1).color;
+    const expectedColor = "rgb(128, 0, 128)";
+
+    assert_equals(computedColor, expectedColor);
+    assert_true(styleExists("h1 { color: purple; }")); // first style sheet
```

Comment:
Somewhat disturbing the CSS parser preserves color names, but I guess that's the way it is.

Reply:
Must admit I'm not entirely sure what you mean by this. Would you prefer an rgb or hex val instead?

Reply:
All I meant is that I was surprised that the parser doesn't normalize these values meaning the browser has to use some additional bits to remember to serialize back out as a name.